groovy/groovy.support/src/org/netbeans/modules/groovy/support/actions/GroovyComputeTestMethods.java

```diff
@@ -53,19 +54,24 @@
                 int classStartColumn = classNode.getColumnNumber();
                 int classOffset = classStartLine > 0 && classStartColumn > 0 ? getOffset(text, classStartLine, classStartColumn) : 0;
                 for (MethodNode methodNode : classNode.getMethods()) {
-                    int startLine = methodNode.getLineNumber();
-                    int startColumn = methodNode.getColumnNumber();
-                    int endLine = methodNode.getLastLineNumber();
-                    int endColumn = methodNode.getLastColumnNumber();
-                    if (startLine > 0 && startColumn > 0 && endLine > 0 && endColumn > 0) {
-                        int startOffset = getOffset(text, startLine, startColumn);
-                        int endOffset = getOffset(text, endLine, endColumn);
-                        result.add(new TestMethodController.TestMethod(classNode.getName(),
-                                new SimplePosition(classOffset),
-                                new SingleMethod(parserResult.getSnapshot().getSource().getFileObject(), methodNode.getName()),
-                                new SimplePosition(startOffset),
-                                new SimplePosition(startOffset),
-                                new SimplePosition(endOffset)));
+                    for (AnnotationNode annotation : methodNode.getAnnotations()) {
+                        if ("org.spockframework.runtime.model.FeatureMetadata".equals(annotation.getClassNode().getName())) {
```

Comment:
Q: does this mean that previously some methods (without FeatureMetadata annotation) could be to the test results although they weren't eligible actually ?

Reply:
Yes, previously all methods of classes extending `spock.lang.Specification` were erroneously marked as Spock test methods (including e.g. `setup()`, `cleanup()`, ...)